x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/hooks/hooks.ts

```diff
@@ -55,3 +59,74 @@ export const useAgentDetailsIngestUrl = (
     };
   }, [services.application, agentId]);
 };
+
+export const useLogEntryUIElements = (logEntry: Immutable<ActivityLogEntry>) => {
+  const iconType = useRef<IconType>('dot');
```

Comment:
I'm trying to understand why you need to use `useRef`. Is there data in these that you need to mutate between renders? (I can't seem to see code that checks the values first)

Feels like this should either be done in the component itself (no hook needed) or just avoid the `useRef()` usage and just do all of the logic inside of the `useMemo()`.

Reply:
cool. If the hook is specific for the Log's component only, I would also move it closer to the component and not put it under the `view/hooks`. Just a suggestion 😄

Reply:
done c4e933a9c5954ce249942ca66bab380c1dfa79e2